### DISCUSSION TITLE
unable to load background

### DISCUSSION THREAD

**Discussion by vaynevayne** [fabricjs/fabric.js#10218] [Q&A] [answered]
Loading the background image with the following code will cause an error. I don't know where the problem is. Can you help me take a look?

stackblitz: https://stackblitz.com/edit/vitejs-vite-sujben?file=src%2FFabric.tsx

```jsx
<ReactFabric
        backgroundImage={
          new FabricImage(
            'https://img-home.csdnimg.cn/images/20240218021830.png',
            {
              width: 500,
              height: 500,
              originX: 'left',
              originY: 'top',
              crossOrigin: 'anonymous',
            }
          )
        }
</ReactFabric>
```

<img width="864" alt="image" src="https://github.com/user-attachments/assets/6e0752bd-f5b6-44ff-b308-58eeed6902ed">

**Comment by asturur** [comment]
You should get a typescript error telling you that FabricImage does not accept a string as first argument.
Aren't you?

**Reply by vaynevayne** [reply]
I can't get any ts error

**Reply by vaynevayne** [reply]
Backgroundimage only accepts imageSource.I don't think there's any need to make it synchronized. This API should support the promise < >I don't think there's any need to make it synchronized. This api should support promise < ImageSourhoweve>，Thinking about improving this API? The point is to emit the background: loader event so that the user can use thebackgroundaimage

**Reply by asturur** [reply, marked as answer]
No the api to load an image is already very simple
Creating an app with react hooks and a canvas is complex enough, loading resources is something the developer has sort out anyway, moving that complexity in fabricJS is not advisable.
We need to remove functionality not cover tasks a developer has to do anyway.
